tutorials/week4-react.md

```diff
@@ -0,0 +1,509 @@
+ */
+function IncrementCounterButton({
+  incrementCount
+}: IncrementCounterButtonProps) {
+  return <button onClick={() => incrementCount()}>increment Count</button>;
+}
+
+/**
+ * Child component accepts a decrementCount callback function as a props.
+ * The callback which is declared inside the parent component triggers a state update.
+ */
+function DecrementCounterButton({
+  decrementCount
+}: DecrementCounterButtonProps) {
+  return <button onClick={() => decrementCount()}>Decrement Count</button>;
+}
+
+function CustomCounterButton({ setCount }: SetCounterButtonProps) {
+  // change this value to see how child passes count value data to parent through
+  // the use of callback function
+
+  const dummyValue = 100;
+
+  return (
+    <button onClick={() => setCount(dummyValue)}>
+      Set Count to {dummyValue}
+    </button>
+  );
+}
+
+/**
+* Parent Component where state and callbacks are maintained. 
+**/
+function Counter() {
+  const [counter, setCounter] = useState(0);
+
+  const decrementCount = () => {
+    if (counter === 0) return;
+    setCounter(counter - 1);
+  };
+
+  const incrementCount = () => {
+    setCounter(counter + 1);
+  };
+
+  const setCount = (value: number) => {
+    setCounter(value);
+  };
+
+  return (
+    <>
+      <CounterContent counter={counter} />
+      <IncrementCounterButton incrementCount={incrementCount} />
+      <DecrementCounterButton decrementCount={decrementCount} />
+      <CustomCounterButton setCount={setCount} />
+    </>
+  );
+}
+```
+
+
+# Handling Events
+
+<a href="https://codesandbox.io/s/nervous-morse-o3pwqm?file=/src/tutorial/HandlingEvents.tsx" target="_blank">view in sandbox</a>
+
+
+-   React impelemnts its own system of handling events that is very similar to handling events on DOM elements. There are some syntax differences:
+    -   React events are named using camelCase, rather than lower case.
+    -   With JSX a function can be passed as an event handler instead of a string.
+
+For example, the HTML
+
+```html
+<button onclick="incrementCounter()">Increment Counter</button>
+```
+
+is slightly different in React:
+
+```html
+<button onclick="{incrementCounter}">Increment Counter</button>
+```
+
+-   React provides an equivalent of each event that you might be familiar with from
+    standard HTML DOM. Events typing can be imported directly from react as shown in the example below:
+
+```ts
+import { MouseEvent } from "react";
+
+function HelloWorld() {
+    const sayHello = (event: MouseEvent) => {
+        event.preventDefault();
+        console.log(`${event.currentTarget.tagName}: says hello, world.`);
+    };
+
+    return (
+        <div>
+            <button onClick={sayHello}>Submit</button>
+        </div>
+    );
+}
+export default HelloWorld;
+```
+
+For the list of all the supported events refer https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forms_and_events/#list-of-event-types
+
```

Comment:
https://react.dev/reference/react-dom/components/common

We could add this reference which contains examples and instances to use events.

Reply:
You should go ahead and update the PR